decimal/decimal_test.go

```diff
@@ -0,0 +1,574 @@
+	if err != nil {
+		t.Fatalf("Decimal insert failed: %s", err)
+	}
+	if resp == nil {
+		t.Fatalf("Response is nil after Replace")
+	}
+	tupleValueIsDecimal(t, resp.Data, number)
+
+	var offset uint32 = 0
+	var limit uint32 = 1
+	resp, err = conn.Select(space, index, offset, limit, IterEq, []interface{}{number})
+	if err != nil {
+		t.Fatalf("Decimal select failed: %s", err.Error())
+	}
+	if resp == nil {
+		t.Fatalf("Response is nil after Select")
+	}
+	tupleValueIsDecimal(t, resp.Data, number)
+
+	resp, err = conn.Delete(space, index, []interface{}{number})
+	if err != nil {
+		t.Fatalf("Decimal delete failed: %s", err)
+	}
+	tupleValueIsDecimal(t, resp.Data, number)
+}
+
+func assertInsert(t *testing.T, conn *Connection, numString string) {
+	number, err := decimal.NewFromString(numString)
+	if err != nil {
+		t.Fatalf("Failed to prepare test decimal: %s", err)
+	}
+
+	resp, err := conn.Insert(space, []interface{}{number})
+	if err != nil {
+		t.Fatalf("Decimal insert failed: %s", err)
+	}
+	if resp == nil {
+		t.Fatalf("Response is nil after Replace")
+	}
+	tupleValueIsDecimal(t, resp.Data, number)
+
+	resp, err = conn.Delete(space, index, []interface{}{number})
+	if err != nil {
+		t.Fatalf("Decimal delete failed: %s", err)
+	}
+	tupleValueIsDecimal(t, resp.Data, number)
+}
+
+func TestInsert(t *testing.T) {
+	if isDecimalSupported == false {
+		t.Skip("Skipping test for Tarantool without decimal support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	samples := append(correctnessSamples, benchmarkSamples...)
+	for _, testcase := range samples {
+		t.Run(testcase.numString, func(t *testing.T) {
+			assertInsert(t, conn, testcase.numString)
+		})
+	}
+}
+
+func TestReplace(t *testing.T) {
+	if isDecimalSupported == false {
+		t.Skip("Skipping test for Tarantool without decimal support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	number, err := decimal.NewFromString("-12.34")
+	if err != nil {
+		t.Fatalf("Failed to prepare test decimal: %s", err)
+	}
+
+	respRep, errRep := conn.Replace(space, []interface{}{number})
+	if errRep != nil {
+		t.Fatalf("Decimal replace failed: %s", errRep)
+	}
+	if respRep == nil {
+		t.Fatalf("Response is nil after Replace")
+	}
+	tupleValueIsDecimal(t, respRep.Data, number)
+
+	respSel, errSel := conn.Select(space, index, 0, 1, IterEq, []interface{}{number})
+	if errSel != nil {
+		t.Fatalf("Decimal select failed: %s", errSel)
+	}
+	if respSel == nil {
+		t.Fatalf("Response is nil after Select")
+	}
+	tupleValueIsDecimal(t, respSel.Data, number)
+}
+
+// runTestMain is a body of TestMain function
+// (see https://pkg.go.dev/testing#hdr-Main).
+// Using defer + os.Exit is not works so TestMain body
+// is a separate function, see
+// https://stackoverflow.com/questions/27629380/how-to-exit-a-go-program-honoring-deferred-calls
+func runTestMain(m *testing.M) int {
+	isLess, err := test_helpers.IsTarantoolVersionLess(2, 2, 0)
```

Comment:
Should we say something about these limitations in README.md or CONTRIBUTING.md?

Reply:
There is Godoc for this.